core/service/src/vault/storage/s3.rs

```diff
@@ -65,6 +65,50 @@ impl S3Storage {
     fn item_key(&self, data_id: &RequestId, data_type: &str) -> String {
         format!("{}/{}/{}", self.prefix, data_type, data_id)
     }
+
+    /// Helper to update cache after successful S3 operation
+    async fn update_cache(&self, key: &str, data: &[u8]) {
+        if let Some(cache) = &self.cache {
+            let cache = Arc::clone(cache);
+            let mut guarded_cache = cache.lock().await;
+            guarded_cache.insert(&self.bucket, key, data);
+        }
+    }
+
+    /// Helper to invalidate cache entry (using poison approach)
+    async fn invalidate_cache(&self, key: &str) {
+        if let Some(cache) = &self.cache {
+            let cache = Arc::clone(cache);
+            let mut guarded_cache = cache.lock().await;
+            guarded_cache.insert(&self.bucket, key, &[]); // Poison with empty data
```

Comment:
just curious: why would we poison the cache and insert an empty element, instead of just deleting the element under that key? What's the benefit?

Reply:
The reason we poison the cache instead of deleting is due to a fundamental API limitation in the 
`StorageCache` implementation, where we have only 2 methods `insert` and `get`. These empty entries are LRU evicted eventually. Although, on a second thought, perhaps it's worth adding additional method and refactor the code a bit, if it's Ok at this stage.

Reply:
I guess we could just implement a delete method, right? `StoraceCache` is our struct. To me it seems a bit more proper. I think it just wasn't implemented previously because it was not needed

Reply:
Agreed. I was just being cautious about avoiding API changes unless absolutely necessary, as we discussed earlier. Changed in [chore: add remove method to StorageCache](https://github.com/zama-ai/kms/pull/42/commits/642af1a65546d52cd520897a2bdec98bbed10a1e)